frontend/lib/editor/common.ts

```diff
@@ -0,0 +1,61 @@
+import { type WritableComputedRef } from 'vue'
+
+export interface ForUse<T> {
+  isRequired: boolean
+  isCompleted: ComputedRef<boolean>
+  value: WritableComputedRef<T>
+}
+
+export enum Validation {
+  NotEmpty = 'NotEmpty',
+  AlphanumericAndDashesAndUnderscores = 'AlphanumericAndDashesAndUnderscores',
+}
+
+export interface EditorField<R, Name extends keyof R> {
+  name: Name
+  label: string
+  isRequired?: boolean
+  validation?: Validation | undefined
+  originalValue: R[Name]
+  currentValue: R[Name]
+}
+
+export const asChange = <R, N extends keyof R>(field: EditorField<R, N>): Partial<R> => {
+  const result: Partial<R> = {}
+  if (field.originalValue === field.currentValue) {
+    return result
+  }
+  result[field.name] = field.currentValue
+  return result
+}
+
+export const asValue = <R, N extends keyof R>(field: EditorField<R, N>): Pick<R, N> => {
+  return { [field.name]: field.currentValue } as unknown as Pick<R, N>
```

Comment:
I spent a few minutes poking around with this. I'm still convinced that it is definitely doable in TypeScript, even if a computer said otherwise. The best I could do is just removing the `unknown` though ([playground link](https://www.typescriptlang.org/play?target=99&jsx=0#code/FAUwHgDg9gTgLgAhAOwK4FsEDUCGAbASwBMc4CpkEBvYBBAOSjgFF0I4BPBAXgQHJGLNpz4AaWggCCeCAAscadCBgEAxpOREAIjgDOskLo1EAqpuW7VsQz37S5CjMrXGd+w8bNELVmIbHAAL7AoJCwiATIcMoAZjiqIAjMRARwsABiBCB4RAA8AEqiDDhKSGDRmroIANYgHFAxCPkAfNQSyCUgAFzFShJ4OABG2T26cCrIAOYSBLr5IACOqAR+RAD8PYNQUHggChIAbvjEpOTIG9jHJGQUCAA+CKjmMZEgRBKwBJOR+Lh4qN0mgBteidAC6ElUqBgfiifwBPXyIPBQRC4Gg8AQVmQYwQenhiV4BSK9DKFSIVVq9UaLQAFC9skQeslUhksjliQxmgBKHoABTU1U59Fa3FaNDofjg0MoVAQQIZOQAdB0lGCeoqiEqoTCUHACQhAniqgLVELClzUUA)), that seemed to work.

```suggestion
  return { [field.name]: field.currentValue } as Pick<R, N>
```

Reply:
Thoroughly improved! PTAL.